packages/@ama-sdk/core/src/plugins/mock-intercept/mock-intercept.angular.ts

```diff
@@ -0,0 +1,76 @@
+import { CUSTOM_MOCK_OPERATION_ID_HEADER, MockInterceptFetchParameters } from './mock-intercept.interface';
+import { MockInterceptRequest } from './mock-intercept.request';
+import { HttpResponse } from '@angular/common/http';
+
+/**
+ * Plugin to mock and intercept the call of SDK
+ *
+ * This plugin should be used only with the MockInterceptRequest Plugin.
+ * It will allow the user to delay the response or to handle the getResponse function provided with the mock (if present).
+ */
+export class MockInterceptAngular implements AngularPlugin {
+
+  constructor(protected options: MockInterceptFetchParameters) {}
+
+  public load(context: AngularPluginContext): PluginObservableRunner<HttpResponse<any>, AngularCall> {
+
+    if (!context.apiClient.options.requestPlugins.some((plugin) => plugin instanceof MockInterceptRequest)) {
+      throw new Error('MockInterceptAngular plugin should be used only with the MockInterceptRequest plugin');
+    }
+
+    return {
+      transform: (call: AngularCall) => {
+        return from((
+          async () => {
+            await this.options.adapter.initialize();
+
+            let responsePromise = call;
+            // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+            if (!context.options.headers || !(context.options.headers instanceof Headers) || !(context.options.headers as Headers).has(CUSTOM_MOCK_OPERATION_ID_HEADER)) {
+              return responsePromise;
+            }
+
+            if (typeof this.options.delayTiming !== 'undefined') {
+              const delayTime = typeof this.options.delayTiming === 'number' ? this.options.delayTiming : await this.options.delayTiming({
+                ...context,
+                fetchPlugins: [],
+                options: context.requestOptions
+              });
+              responsePromise = responsePromise.pipe(delay(delayTime));
+            }
+
+            // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+            const operationId = (context.options.headers as Headers).get(CUSTOM_MOCK_OPERATION_ID_HEADER)!;
+            try {
+              const mock = this.options.adapter.getLatestMock(operationId);
+
+              if (!mock.getResponse) {
+                return responsePromise;
+              }
+
+              const response = mock.getResponse();
+              return responsePromise.pipe(
+                mergeMap(async (res) => {
+                  const body = await response.json();
+                  const responseCloned = res.clone();
+                  return new HttpResponse<any>({
+                    ...responseCloned,
+                    body,
+                    url: responseCloned.url || undefined
+                  });
+                })
+              );
+
+            } catch {
+              (context.logger || console).error(`Failed to retrieve the latest mock for Operation ID ${operationId}, fallback to default mock`);
```

Comment:
Is it a fallback to default mock or to default call (the real call)?

Reply:
Copy from original fetch plugin but you're right, I will fix both of them